src/backend/distributed/commands/function.c

```diff
@@ -213,6 +220,33 @@ create_distributed_function(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * ErrorIfAnyNodeDoesNotHaveMetadataSynced throws error if any
+ * of the worker nodes does not have the metadata.
+ */
+static void
+ErrorIfAnyNodeDoesNotHaveMetadataSynced(void)
+{
+	List *workerNodeList =
+		ActivePrimaryNonCoordinatorNodeList(ShareLock);
+
+	WorkerNode *workerNode = NULL;
+	foreach_ptr(workerNode, workerNodeList)
+	{
+		if (!workerNode->hasMetadata)
```

Comment:
Do we need to check metadataSynced as well? If not, should we update the name of the function to ErrorIfAnyNodeDoesNotHaveMetadata (including the comment we've added while calling that function) ?

Reply:
if `metadataSynced = false`, `ErrorIfAnyMetadataNodeOutOfSync` throws an error at the low level - while sending command(s). So, I'd rather not add that check here as well. 

I can change the name of the function though.